pkg/gits/github.go

```diff
@@ -445,7 +445,13 @@ func (p *GitHubProvider) CreateWebHook(data *GitWebHookArguments) error {
 	}
 
 	log.Logger().Infof("Creating GitHub webhook for %s/%s for url %s", util.ColorInfo(owner), util.ColorInfo(repo), util.ColorInfo(webhookUrl))
-	_, _, err = p.Client.Repositories.CreateHook(p.Context, owner, repo, hook)
+	_, resp, err := p.Client.Repositories.CreateHook(p.Context, owner, repo, hook)
+	if err != nil {
+		if resp.StatusCode == 404 || resp.StatusCode == 403 {
+			return errors.Wrap(err, "Permission denied")
```

Comment:
+1, maybe we can provide a bit more context `errors.Wrapf(err, "unable to create webhook for %s/%s - permission denied", owner, repo)`.
This way when you get the error you have a bit more context and you don't need to figure out what caused a 404.

Do you mind adding some tests as well?

Reply:
Hey, @hferentschik how can I mock a Github provider for the test case?

Reply:
On the second thought, a `MockGitProvider` won't cut it here. You would need to create the `GitHubProvider` test instance and set its `Client` instance to something you can control, aka some test server from which you can return what you need/want (opposed to talking to GitHub directly).

The way to do this is shown here - https://github.com/google/go-github/blob/master/github/github_test.go#L34
You need this _setup_ function as a helper for your tests. It allows you to spin off an HTTP server for which you can control what requests to handle and how to respond. 

You can see how this is used here - https://github.com/google/go-github/blob/master/github/repos_hooks_test.go

I think it would make for a nice test and open the doors for better testing of `GitHubProvider`. Once the helper method is there we can write more tests using it.

WDYT, do you want to give it a go? Happy to help in case you need some help.

Reply:
Sure, Can I please have another PR for that? It seems it may take some time to learn it.